profile/src/v100/app_preferences/gateways/gateway.rs

```diff
@@ -0,0 +1,277 @@
+use crate::prelude::*;
+use std::ops::Deref;
+
+/// A gateway to some Radix Network, which is a high level REST API which clients (wallets) can
+/// consume in order to query asset balances and submit transactions.
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+    Hash,
+    derive_more::Display,
+    derive_more::Debug,
+    uniffi::Record,
+)]
+#[display("{}", self.url.to_string())]
+#[debug("{}: {}", self.network.display_description, self.url.to_string())]
+pub struct Gateway {
+    /// The Radix network the API is a Gateway to.
+    pub network: RadixNetwork,
+
+    /// The URL to the gateways API endpoint
+    pub url: Url,
+}
+
+impl Identifiable for Gateway {
+    type ID = Url;
+
+    fn id(&self) -> Self::ID {
+        self.url.clone()
+    }
+}
+
+impl Default for Gateway {
+    fn default() -> Self {
+        Self::mainnet()
+    }
+}
+
+impl Gateway {
+    pub fn new(
+        url: String,
+        id: NetworkID,
+    ) -> Result<Arc<Self>, crate::CommonError> {
```

Comment:
FIXME: This is wrong, should not involve `Arc` at all, remnant of old code... I will fix